Sources/Commands/Utilities/PluginDelegate.swift

```diff
@@ -389,8 +389,19 @@ final class PluginDelegate: PluginInvocationDelegate {
             throw StringError("could not find a target named “\(targetName)”")
         }
 
+        // FIXME: This is currently necessary because `target(for:destination:)` can
+        // produce a module that is targeting host when `targetName`` corresponds to
+        // a macro, plugin, or a test. Ideally we'd ask a build system for a`BuildSubset`
+        // and get the destination from there but there are other places that need
+        // refactoring in that way as well.
+        let buildParameters = if target.buildTriple == .tools {
```

Comment:
@rauhul FYI, I also fixed problem in symbol graph creation.

Reply:
Thank you!

Reply:
Feel free to close #7629 when this merges!

Reply:
Will do!